libs/hdf-converters/src/converters-from-hdf/splunk/reverse-splunk-mapper.ts

```diff
@@ -343,66 +344,64 @@ export class FromHDFToSplunkMapper extends FromAnyBaseConverter {
   }
 
   async uploadSplunkData(
-    targetIndex: splunkjs.Index,
+    config: SplunkConfig,
+    targetIndex: any,
     splunkData: SplunkData
   ): Promise<void> {
     return new Promise((resolve, reject) => {
+      const hostname = config.port
+        ? `${config.scheme}://${config.host}:${config.port}`
+        : `${config.scheme}://${config.host}:8089`;
+      this.axiosInstance.defaults.params['sourcetype'] = MAPPER_NAME;
```

Comment:
I'm not sure if I understand what you're trying to say here. Feel free to correct me if I'm not answering your question.

Is this an issue with these variables being constant? If so, wouldn't that not matter since the values are handled by the garbage collector the moment the function call ends and they leave scope? By the time we enter another checkSplunk call, the variables from the previous function call would've already been flushed out with the calll stack change. And since these variables are generated off of whatever is passed in for `config`, they're independent of each other anyways.

Reply:
I think I lost track of overarching scope and found a potential shape of a problem and made a comment about it.  Basically my concern was that we were setting state on a global object within some random function that could've been potentially called more than once.  But now I'm seeing it doesn't actually get called more than once probably lol.